app/lib/actions/candlepin/import_pool_handler.rb

```diff
@@ -35,14 +37,19 @@ def reindex_consumer(message_handler)
         subscription_facet = message_handler.subscription_facet
         sub_status = message_handler.sub_status
         uuid = message_handler.consumer_uuid
+
         if subscription_facet && sub_status
           @logger.debug "re-indexing content host #{subscription_facet.host.name}"
-          subscription_facet.update_subscription_status(sub_status)
-          subscription_facet.update_compliance_reasons(message_handler.consumer_reasons)
+          if message_handler.subject.match(/system_purpose_compliance\.created/)
+            subscription_facet.update_purpose_status(message_handler.event_data)
+          else
+            subscription_facet.update_subscription_status(sub_status)
+            subscription_facet.update_compliance_reasons(message_handler.consumer_reasons)
+          end
```

Comment:
This isn't a huge method, but would it make sense to break this up into 'reindex_subscription_status' and 'reindex_purpose_status' methods?  Checking the message subject a 2nd time here (when it was already checked above) smells a little to me.    Its not a huge deal and if you want to leave it as is its okay, thought i'd mention it.

Reply:
Yeah it stinks :+1: